### DISCUSSION TITLE
Cannot start configuration is not valid

### DISCUSSION THREAD

**Discussion by kimbo1st** [cytech/Home-Assistant-wundergroundpws#171] [General]
Have this error,and don’t know how to solve.
Used the sample config file.

`Logger: homeassistant.components.homeassistant
Source: components/homeassistant/__init__.py:170 
Integration: Home Assistant Core Integration (documentation, issues) 
First occurred: 00:06:43 (1 occurrences) 
Last logged: 00:06:43

The system cannot restart because the configuration is not valid: Platform error weather.wundergroundpws - cannot import name 'WundergroundPWSUpdateCoordinator' from 'custom_components.wundergroundpws' (/config/custom_components/wundergroundpws/__init__.py) Invalid config for [sensor.wundergroundpws]: required key not provided @ data['api_key']. Got None required key not provided @ data['numeric_precision']. Got None required key not provided @ data['pws_id']. Got None value must be one of ['dewpt', 'elev', 'heatIndex', 'humidity', 'neighborhood', 'obsTimeLocal', 'precipRate', 'precipTotal', 'precip_1d', 'precip_2d', 'precip_3d', 'precip_4d', 'precip_5d', 'precip_chance_1d', 'precip_chance_2d', 'precip_chance_3d', 'precip_chance_4d', 'precip_chance_5d', 'pressure', 'solarRadiation', 'stationID', 'temp', 'temp_high_1d', 'temp_high_2d', 'temp_high_3d', 'temp_high_4d', 'temp_high_5d', 'temp_low_1d', 'temp_low_2d', 'temp_low_3d', 'temp_low_4d', 'temp_low_5d', 'today_summary', 'uv', 'weather_1d', 'weather_1n', 'weather_2d', 'weather_2n', 'weather_3d', 'weather_3n', 'weather_4d', 'weather_4n', 'weather_5d', 'weather_5n', 'windChill', 'windGust', 'windSpeed', 'wind_1d', 'wind_2d', 'wind_3d', 'wind_4d', 'wind_5d', 'winddir'] @ data['monitored_conditions'][12]. Got 'precip_chance_1n' value must be one of ['dewpt', 'elev', 'heatIndex', 'humidity', 'neighborhood', 'obsTimeLocal', 'precipRate', 'precipTotal', 'precip_1d', 'precip_2d', 'precip_3d', 'precip_4d', 'precip_5d', 'precip_chance_1d', 'precip_chance_2d', 'precip_chance_3d', 'precip_chance_4d', 'precip_chance_5d', 'pressure', 'solarRadiation', 'stationID', 'temp', 'temp_high_1d', 'temp_high_2d', 'temp_high_3d', 'temp_high_4d', 'temp_high_5d', 'temp_low_1d', 'temp_low_2d', 'temp_low_3d', 'temp_low_4d', 'temp_low_5d', 'today_summary', 'uv', 'weather_1d', 'weather_1n', 'weather_2d', 'weather_2n', 'weather_3d', 'weather_3n', 'weather_4d', 'weather_4n', 'weather_5d', 'weather_5n', 'windChill', 'windGust', 'windSpeed', 'wind_1d', 'wind_2d', 'wind_3d', 'wind_4d', 'wind_5d', 'winddir'] @ data['monitored_conditions'][14]. Got 'precip_chance_2n' value must be one of ['dewpt', 'elev', 'heatIndex', 'humidity', 'neighborhood', 'obsTimeLocal', 'precipRate', 'precipTotal', 'precip_1d', 'precip_2d', 'precip_3d', 'precip_4d', 'precip_5d', 'precip_chance_1d', 'precip_chance_2d', 'precip_chance_3d', 'precip_chance_4d', 'precip_chance_5d', 'pressure', 'solarRadiation', 'stationID', 'temp', 'temp_high_1d', 'temp_high_2d', 'temp_high_3d', 'temp_high_4d', 'temp_high_5d', 'temp_low_1d', 'temp_low_2d', 'temp_low_3d', 'temp_low_4d', 'temp_low_5d', 'today_summary', 'uv', 'weather_1d', 'weather_1n', 'weather_2d', 'weather_2n', 'weather_3d', 'weather_3n', 'weather_4d', 'weather_4n', 'weather_5d', 'weather_5n', 'windChill', 'windGust', 'windSpeed', 'wind_1d', 'wind_2d', 'wind_3d', 'wind_4d', 'wind_5d', 'winddir'] @ data['monitored_conditions'][16]. Got 'precip_chance_3n' value must be one of ['dewpt', 'elev', 'heatIndex', 'humidity', 'neighborhood', 'obsTimeLocal', 'precipRate', 'precipTotal', 'precip_1d', 'precip_2d', 'precip_3d', 'precip_4d', 'precip_5d', 'precip_chance_1d', 'precip_chance_2d', 'precip_chance_3d', 'precip_chance_4d', 'precip_chance_5d', 'pressure', 'solarRadiation', 'stationID', 'temp', 'temp_high_1d', 'temp_high_2d', 'temp_high_3d', 'temp_high_4d', 'temp_high_5d', 'temp_low_1d', 'temp_low_2d', 'temp_low_3d', 'temp_low_4d', 'temp_low_5d', 'today_summary', 'uv', 'weather_1d', 'weather_1n', 'weather_2d', 'weather_2n', 'weather_3d', 'weather_3n', 'weather_4d', 'weather_4n', 'weather_5d', 'weather_5n', 'windChill', 'windGust', 'windSpeed', 'wind_1d', 'wind_2d', 'wind_3d', 'wind_4d', 'wind_5d', 'winddir'] @ data['monitored_conditions'][18]. Got 'precip_chance_4n' value must be one of ['dewpt', 'elev', 'heatIndex', 'humidity', 'neighborhood', 'obsTimeLocal', 'precipRate', 'precipTotal', 'precip_1d', 'precip_2d', 'precip_3d', 'precip_4d', 'precip_5d', 'precip_chance_1d', 'precip_chance_2d', 'precip_chance_3d', 'precip_chance_4d', 'precip_chance_5d', 'pressure', 'solarRadiation', 'stationID', 'temp', 'temp_high_1d', 'temp_high_2d', 'temp_high_3d', 'temp_high_4d', 'temp_high_5d', 'temp_low_1d', 'temp_low_2d', 'temp_low_3d', 'temp_low_4d', 'temp_low_5d', 'today_summary', 'uv', 'weather_1d', 'weather_1n', 'weather_2d', 'weather_2n', 'weather_3d', 'weather_3n', 'weather_4d', 'weather_4n', 'weather_5d', 'weather_5n', 'windChill', 'windGust', 'windSpeed', 'wind_1d', 'wind_2d', 'wind_3d', 'wind_4d', 'wind_5d', 'winddir'] @ data['monitored_conditions'][20]. Got 'precip_chance_5n' value must be one of ['dewpt', 'elev', 'heatIndex', 'humidity', 'neighborhood', 'obsTimeLocal', 'precipRate', 'precipTotal', 'precip_1d', 'precip_2d', 'precip_3d', 'precip_4d', 'precip_5d', 'precip_chance_1d', 'precip_chance_2d', 'precip_chance_3d', 'precip_chance_4d', 'precip_chance_5d', 'pressure', 'solarRadiation', 'stationID', 'temp', 'temp_high_1d', 'temp_high_2d', 'temp_high_3d', 'temp_high_4d', 'temp_high_5d', 'temp_low_1d', 'temp_low_2d', 'temp_low_3d', 'temp_low_4d', 'temp_low_5d', 'today_summary', 'uv', 'weather_1d', 'weather_1n', 'weather_2d', 'weather_2n', 'weather_3d', 'weather_3n', 'weather_4d', 'weather_4n', 'weather_5d', 'weather_5n', 'windChill', 'windGust', 'windSpeed', 'wind_1d', 'wind_2d', 'wind_3d', 'wind_4d', 'wind_5d', 'winddir'] @ data['monitored_conditions'][56]. Got 'windDirectionName'. (See ?, line ?). `

**Comment by cytech** [comment]
my guess is you installed v2.xx using the v1.xx instructions (or vice versa)
v2.xx instructions https://github.com/cytech/Home-Assistant-wundergroundpws/tree/v2.X.X

**Reply by kimbo1st** [reply]
This solved it.. thanks!
